### PR TITLE
Support secured probe endpoints

### LINKED ISSUE #236: Support secured probe endpoints
Support the ability to use probe endpoints that are secured with basic auth, configured via deployer properties and credentials stored as a secret. The same credentials and auth type would be used for both readiness and liveness probes. Relates to #232

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #236

### CODE REVIEW

**Comment by chrisjs** [comment]
for testing from dataflow, i used the `latest` tag since those images have security functionality that can be configured for testing. http has security off by default but can be enabled with ` --spring.security.user.name=user --spring.security.user.password=pass --http.enable-security=true`. custom actuator endpoints are defined as the current defaults are boot 1.x paths.

info on creating the `probeCredentialsSecret` can be found in the referenced doc issue

```
app register --name log --type sink --uri docker://springcloudstream/log-sink-rabbit:latest
app register --name http --type source --uri docker://springcloudstream/http-source-rabbit:latest

stream create --name authtest --definition "http --management.endpoints.web.exposure.include=health,info | log --management.endpoints.web.exposure.include=health,info --spring.security.user.name=user --spring.security.user.password=pass"

stream deploy authtest --properties "deployer.http.kubernetes.livenessProbePath=/actuator/health,deployer.http.kubernetes.readinessProbePath=/actuator/info,deployer.log.kubernetes.livenessProbePath=/actuator/health,deployer.log.kubernetes.readinessProbePath=/actuator/info,deployer.log.kubernetes.probeCredentialsSecret=myprobesecret"
```

after some time and the apps are deployed `kubectl describe <pod>` can be used to verify there are no probe failures, or use invalid credentials in a secret to see it fail

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java [hunk 1]
```diff
@@ -444,5 +451,16 @@ private String getDeploymentServiceAccountName(AppDeploymentRequest request) {
 		}
 
 		return deploymentServiceAccountName;
```
Comment (tzolov) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java]: Looks like a missing brackets after the return leading to compilation failure.

Comment (chrisjs) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java]: rebase error.. fixed

**Comment by tzolov** [comment]
LGTM, ready to merge

**Comment by tzolov** [comment]
Merged at c4c39ed

**Pull request closed without merging**